dragnet/model_training.py

```diff
@@ -93,8 +94,18 @@ def make_features_from_data(self, data, model, training_or_test='training',
                 # document is too short -- has < blocks.
                 # skip it
                 continue
-            features = np.vstack((features, training_features))
             this_labels, this_weight, this_tokens = self._get_labels(content, comments)
+
+            if (training_features.shape[0] != len(this_weight) or
+                    len(this_labels) != len(this_weight)):
+                print('\nskipping file because of array shape mismatch...')
+                print('len(blocks) =', len(blocks))
+                print('len(this_labels) =', len(this_labels))
+                print('len(this_weight) =', len(this_weight))
+                # raise ValueError("Number of features, labels and weights do not match!")
```

Comment:
Let's continue to raise this an error.  By trapping it, we could mask another larger significant problem and create a much smaller or corrupted data set.  Someone who is retraining without being familiar with the code details is likely to not notice it, so let's make it explicit when it is ignored.